GOVERNANCE.md

```diff
@@ -0,0 +1,126 @@
+* Concise syntax (minimal redundancy in code)
+* Stable code base (easy for users to upgrade to new data.table, and compatible with old R versions)
+* Comprehensive and accessible documentation
+* Clear error and warning messages
+
+To prioritize developer time, we define what is in and out of scope.
+The current scope of package functionality includes:
+* Data manipulation and analysis 
+    * reshaping/pivoting
+    * aggregation/summarizing
+    * subsetting rows
+    * all sorts of joins
+    * adding/updating/deleting columns
+* Reading/writing of data from/to many file formats
+
+Functionality that is out of scope:
+* plotting/graphics (like ggplot2)
+* manipulating data stored on disk (or remote SQL DB) rather than in memory (like sqldf / dbplyr)
+* machine learning / modeling (like mlr3)
+* regular expression builders (like rex and nc packages) 
+
+# Roles 
+
+## Contributor
+
+* Definition: a user who has written/commented at least one issue, worked to label/triage issues, written a blog post, given a talk, etc. 
+* How this role is recognized: there is no central list of contributors / no formal recognition for contributors.
+
+## Project member
+
+* Definition: some one who has submitted at least one PR that has been merged into master. 
+* How to obtain this role: any user can become a member by submitting a PR, then having it reviewed and merged into master. Contributors who have written issues should be encouraged to submit their first PR to become a project member.
+* How this role is recognized: Members are credited via role="ctb" in DESCRIPTION (so they appear in Author list on CRAN), and they are added to https://github.com/orgs/Rdatatable/teams/project-members so they can create new branches in the Rdatatable/data.table GitHub repo. They also appear on https://github.com/Rdatatable/data.table/graphs/contributors (Contributions to master, excluding merge commits).
+
+## Reviewer
+
+* Definition: a member who has volunteered to do code reviews for some features/files. 
+* How to obtain this role: after one or more significant PRs to a given file, a member should be invited to add their name as a reviewer of that file in CODEOWNERS, and after that is merged into master, then they are considered a reviewer. 
+* How this role is recognized: same credit in DESCRIPTION as a regular member, role="ctb" (so they appear in Author list on CRAN). 
+* Note: having your name in CODEOWNERS does not give any special permission, but it does mean that you will be notified whenever there is a new PR with changes to that file. 
+
+## Committer
+
+* Definition: permission to commit to, and merge PRs into, master branch. 
+* How to obtain this role: after a reviewer has a consistent history of careful reviews of others' PRs, then a current GitHub maintainer should ask all other current GitHub maintainers if they approve promoting the reviewer to GitHub maintainer, and it should be done if there is general consensus/agreement. 
```

Comment:
Should we formalize "retirement", if full consensus is needed to grow the "pool" of committers? Or have a policy of "non-response is tacit agreement"? Complexity of getting consensus o/w risks getting out of control.

Reply:
I think "non-response is tacit agreement" is what I intended, we should clarify